app/models/document.rb

```diff
@@ -206,10 +206,12 @@ def save!
       presented_document = DocumentPresenter.new(self)
       presented_links = DocumentLinksPresenter.new(self)
 
-      handle_remote_error do
+      success = handle_remote_error do
         Services.publishing_api.put_content(self.content_id, presented_document.to_json)
         Services.publishing_api.patch_links(self.content_id, presented_links.to_json)
       end
+
+      raise RecordNotSaved unless success
     else
       raise RecordNotSaved
     end
```

Comment:
Should we consider adding a non-error raising version of this method, too?

```
def save
  save!
  true
rescue RecordNotSaved => e
  # do something with e.message ???
  false
end
```

It seems like we're rescuing from `RecordNotSaved`. It might be cleaner to have a regular `#save` method so that we're not raise exceptions for flow control. What do you think? Maybe a separate refactor?